codec/unknownproto/unknown_fields.go

```diff
@@ -441,3 +441,70 @@ func (d DefaultAnyResolver) Resolve(typeURL string) (proto.Message, error) {
 	}
 	return reflect.New(mt.Elem()).Interface().(proto.Message), nil
 }
+
+func findWireTypeFromFieldDescriptorProtoType(fieldType descriptorpb.FieldDescriptorProto_Type) protowire.Type {
```

Comment:
The name of the function can be improved to something more succinct like "toProtoWireType", but most importantly, I highly advise using the constants as they are from protowire https://pkg.go.dev/google.golang.org/protobuf/encoding/protowire#Type and not using magic numbers as it is quite challenging to debug, maintain and fix bugs; please do the following:

```go

func toProtoWireType(fieldType descriptorpb.FieldDescriptorProto_Type) protowire.Type {
	switch fieldType {
	...
	case descriptorpb.FieldDescriptorProto_TYPE_INT64,
             descriptorpb.FieldDescriptorProto_TYPE_UINT64,
             descriptorpb.FieldDescriptorProto_TYPE_INT32,
             descriptorpb.FieldDescriptorProto_TYPE_UINT32,
             descriptorpb.FieldDescriptorProto_TYPE_BOOL,
             descriptorpb.FieldDescriptorProto_TYPE_ENUM,
             descriptorpb.FieldDescriptorProto_TYPE_SINT32,
             descriptorpb.FieldDescriptorProto_TYPE_SINT64:
	return protowire.VarintType

   case descriptorpb...:
   ...
	}

	panic(fmt.Sprintf("Unhandled type: %s", fieldType))
```